src/PrpShop/PRP/Surface/QMipmap.cpp

```diff
@@ -48,17 +50,39 @@ static void setExportDir(const QString& filename)
     settings.setValue("ExportDir", dir.absolutePath());
 }
 
+std::unique_ptr<unsigned char[]> getTextureData(plMipmap *tex, size_t level=0)
+{
+    if (level >= tex->getNumLevels())
+        level = tex->getNumLevels() - 1;
```

Comment:
Should probably also sanity check that `getNumLevels() > 0`...  A Q_ASSERT() should be sufficient since I think the callers already check that case below.